bio/deeptools/bampefragmentsize/wrapper.py

```
@@ -0,0 +1,52 @@
+# If no labels are provided, use the basename of the bam file
+sample_label = snakemake.params.get("labels", "")
+if not sample_label:
+    sample_label = [os.path.basename(bam.replace(".bam", "")) for bam in bam_files]
+
+# Check if the number of labels is equal to the number of bam files
+assert len(sample_label) == len(
+    bam_files
+), "Number of labels must be equal to the number of bam files"
+
+# Check output format
+out_format = Path(snakemake.output.hist).suffix
```

Ruff (0.8.2)

Undefined name `snakemake`

(F821)